### PR TITLE
Fix revoke of implicit grants

### PR DESCRIPTION
**Pull request by pufit**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the situation when a user can't run `REVOKE ALL ON *.*` because of implicit grants in the target access entity. 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 79e165f79c7e08d17f19d49deb0b251b9eb1e2b0 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72872/79e165f79c7e08d17f19d49deb0b251b9eb1e2b0/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/f752c0b89d1a84fc92bfbb7fc510a092b5aa4c84/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/Access/InterpreterGrantQuery.cpp [hunk 1]
```diff
@@ -190,7 +190,7 @@ namespace
         /// REVOKE SELECT ON system.* FROM user2;
         ///
         /// the query `REVOKE SELECT ON *.* FROM user1` executed by user2 should succeed.
-        if (current_user_access.getAccessRights()->containsWithGrantOption(access_to_revoke))
+        if (current_user_access.getAccessRightsWithImplicit()->containsWithGrantOption(access_to_revoke))
```
Comment (tuanpach) [inline src/Interpreters/Access/InterpreterGrantQuery.cpp:193]: Could you kindly explain the difference between `access` and `access_with_implicit`?

Comment (pufit) [reply src/Interpreters/Access/InterpreterGrantQuery.cpp:193]: We have some implicit grants in our RBAC. For example, when a user has `SELECT` on a table, they should also be able to execute `SHOW CREATE TABLE`. Thus, we add a `SHOW` grant implicitly under the hood.

```

SHOW GRANTS FOR test

Query id: e8d80ef0-f85f-4e75-bd78-e9eeb44ae26c

   ┌─GRANTS FOR test───────────────────────────────┐
1. │ GRANT SELECT ON *.* TO test WITH GRANT OPTION │
   └───────────────────────────────────────────────┘

1 row in set. Elapsed: 0.002 sec. 

pufit-MBP.attlocal.net :) SHOW GRANTS FOR test WITH IMPLICIT

SHOW GRANTS FOR test WITH IMPLICIT

Query id: b1fb45ac-823e-467e-aea3-f64dc0c58060

    ┌─GRANTS FOR test WITH IMPLICIT────────────────────────────────────────────────────────────┐
 1. │ GRANT SHOW DATABASES, SHOW TABLES, SHOW COLUMNS, SELECT ON *.* TO test WITH GRANT OPTION │
    └──────────────────────────────────────────────────────────────────────────────────────────┘

19 rows in set. Elapsed: 0.002 sec. 

pufit-MBP.attlocal.net :)

```

Comment (tuanpach) [reply src/Interpreters/Access/InterpreterGrantQuery.cpp:193]: Got it. Thanks

**Review by tuanpach — APPROVED** [review]

**Pull request merged**